### PR TITLE
Location ID lookup

### PR DESCRIPTION
**Pull request by aaron-russell**
### Work done
- Added new endpoints for searching by postcode or location id
- Added new screen to search by location id or postcode

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

#### Note
- Changes for parent flow not added

### CODE REVIEW

File: server/test/unit/routes/locations/postcode/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    updatedat: new Date('2019-11-26T11:46:16.537Z'),
+  },
+};
+const establishment = [
+  {
+    locationId: '1-2111759818',
+  },
+];
+
+describe.only('locations route', () => {
```
Comment (joannafawl) [inline server/test/unit/routes/locations/postcode/index.spec.js:27]:
```suggestion
describe('locations route', () => {
```

File: server/routes/locations/postcodeOrLocationID/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const getLocationsPCRoute = require('../postcode');
+
+const getLocationsByPostcodeOrLocationID = async (req, res, matching) => {
+  // this sinitises the postcode but also checks it's a postcode
+  let postcodeOrLocationID = pCodeCheck.sanitisePostcode(req.params.postcodeOrLocationID);
+
+  if (postcodeOrLocationID !== null) {
+    return await getLocationsPCRoute.getLocationsByPostcode(req, res, matching, postcodeOrLocationID);
+  } else {
+    return await getLocationsRoute.getLocations(req, res, matching, req.params.postcodeOrLocationID);
```
Comment (Richard-Pentecost) [inline server/routes/locations/postcodeOrLocationID/index.js:15]: Can this ever be reached? Is it possible for there be a req.params.postcodeOrLocationID, but for the sanitised version (postcodeOrLocationID) to return null?

File: server/routes/locations/locationID/index.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+router.route('/:locationId');
+
+router.route('/:locationId').get(async function (req, res) {
+  const locationID = req.params.locationId;
+  await getLocations(req, res, false, locationID);
+});
+
+router.get('/matching/:locationId', Authorization.isAuthorised);
+// GET Location API by locationId
+router.route('/matching/:locationId').get(async function (req, res) {
```
[the remark this answers is not in the document]
Comment (Richard-Pentecost) [reply server/routes/locations/locationID/index.js:70]: What's the purpose of line 80?

File: server/routes/locations/postcodeOrLocationID/index.js (hunk 2 above)
Comment (Richard-Pentecost) [inline server/routes/locations/postcodeOrLocationID/index.js:15]: Can this code ever be reached? Can req.params.postcodeOrLocationID be present, but postcodeOrLocationID be null?

**Review by Richard-Pentecost — COMMENTED** [review]
Just had a couple of questions

**Review by rramgolam — APPROVED** [review]

**Pull request merged**